src/awala.spec.ts

```diff
@@ -34,206 +44,243 @@ describe('awala', () => {
+      signature: 'test',
+      awalaPda: 'test',
+      memberId: MEMBER_MONGO_ID,
+    };
+    await memberBundleRequestModel.create(data);
 
-      const dbResult = await memberBundleRequestModel.findOne({
-        publicKeyId: MEMBER_PUBLIC_KEY_MONGO_ID,
-      });
-      expect(result).toStrictEqual({
-        didSucceed: true,
-      });
-      expect(dbResult).not.toBeNull();
-      expect(dbResult!.awalaPda.toString('base64')).toBe(AWALA_PDA);
-      expect(dbResult!.signature.toString('base64')).toBe(SIGNATURE);
-      expect(dbResult!.memberBundleStartDate).toBeDate();
-      expect(dbResult!.memberBundleStartDate.toISOString()).toBe(futureTimestamp);
-      expect(mockLogging.logs).toContainEqual(
-        partialPinoLog('info', 'Member bundle request created', {
-          publicKeyId: MEMBER_PUBLIC_KEY_MONGO_ID,
-        }),
-      );
+    await createMemberBundleRequest(methodInput, serviceOptions);
+    const dbResult = await memberBundleRequestModel.findOne({
+      publicKeyId: memberPublicKey._id.toString(),
     });
+    expect(dbResult).not.toBeNull();
+    expect(dbResult!.memberId).toBe(MEMBER_MONGO_ID);
+    expect(dbResult!.awalaPda.toString('base64')).toBe(AWALA_PDA);
+    expect(dbResult!.signature.toString('base64')).toBe(SIGNATURE);
+    expect(dbResult!.memberBundleStartDate).toBeDate();
+    expect(dbResult!.memberBundleStartDate.toISOString()).toBe(futureTimestamp);
+  });
 
-    test('Member bundle data should be updated', async () => {
-      const data = {
-        publicKeyId: MEMBER_PUBLIC_KEY_MONGO_ID,
-        memberBundleStartDate: new Date(),
-        signature: 'test',
-        awalaPda: 'test',
-      };
-      await memberBundleRequestModel.create(data);
-
-      await createMemberBundleRequest(methodInput, serviceOptions);
-      const dbResult = await memberBundleRequestModel.findOne({
-        publicKeyId: MEMBER_PUBLIC_KEY_MONGO_ID,
-      });
-      expect(dbResult).not.toBeNull();
-      expect(dbResult!.awalaPda.toString('base64')).toBe(AWALA_PDA);
-      expect(dbResult!.signature.toString('base64')).toBe(SIGNATURE);
-      expect(dbResult!.memberBundleStartDate).toBeDate();
-      expect(dbResult!.memberBundleStartDate.toISOString()).toBe(futureTimestamp);
+  test('Existing data should not create new entry', async () => {
+    await memberBundleRequestModel.create({
+      ...methodInput,
+      memberId: MEMBER_MONGO_ID,
     });
 
-    test('Existing data should not create new entry', async () => {
-      await memberBundleRequestModel.create(methodInput);
+    await createMemberBundleRequest(methodInput, serviceOptions);
 
-      await createMemberBundleRequest(methodInput, serviceOptions);
+    const countResult = await memberBundleRequestModel.count({
+      publicKeyId: memberPublicKey._id.toString(),
+    });
+    expect(countResult).toBe(1);
+  });
 
-      const countResult = await memberBundleRequestModel.count({
+  test('Invalid public key id should be refused', async () => {
```

Comment:
"invalid" is too ambiguous.

```suggestion
  test('Unknown public key id should be refused', async () => {
```

Reply:
That works too